src/state/slices/searchSlice.tsx

```diff
@@ -0,0 +1,51 @@
+import { createSelector, createSlice } from '@reduxjs/toolkit';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from '../store';
+import { DateRange } from '../../app.types';
+import { selectPage, selectSort, selectResultsPerPage } from './columnsSlice';
+
+// Define a type for the slice state
+interface SearchState {
+  dateRange: DateRange;
+}
+
+// Define the initial state using that type
+export const initialState = {
+  dateRange: {},
+} as SearchState;
+
+export const searchSlice = createSlice({
+  name: 'search',
+  // `createSlice` will infer the state type from the `initialState` argument
+  initialState,
+  reducers: {
+    // Use the PayloadAction type to declare the contents of `action.payload`
+    changeDateRange: (
+      state,
+      action: PayloadAction<{ range: keyof DateRange; date?: string }>
+    ) => {
+      const { range, date } = action.payload;
+      state.dateRange[range] = date;
+    },
```

Comment:
Why isn't this part of the columns reducer?

Reply:
Because the datePicker component is not part of the table component - it instead is going to be part of the search component which will have other things the user can search on like experiment ID (OG10), shot ID (OG11), and timeframe (OG74). The redux-toolkit docs said a good way to approaching slices is to slice them according to the main component which uses them, hence why I separated `dateRange`. Both the state from `TableSlice` and `SearchSlice` will have to be passed to the records endpoint, but I figured it was a nice way to split things to prevent a slice from getting too large.

Reply:
That makes sense and I agree with your last point!